calyx-py/calyx/builder.py

```diff
@@ -404,24 +432,21 @@ def infer_width(expr):
             return inst.args[0]
         elif port_name == "write_en":
             return 1
-    elif prim == "std_add":
+    elif prim in ("std_add", "std_lt", "std_eq"):
         if port_name == "left" or port_name == "right":
             return inst.args[0]
-    elif prim == "std_mem_d1":
+    elif prim == "std_mem_d1" or prim == "seq_mem_d1":
         if port_name == "write_en":
             return 1
         elif port_name == "addr0":
             return inst.args[2]
         elif port_name == "in":
             return inst.args[0]
-    elif (
-        prim == "std_mult_pipe"
-        or prim == "std_smult_pipe"
-        or prim == "std_mod_pipe"
-        or prim == "std_smod_pipe"
-        or prim == "std_div_pipe"
-        or prim == "std_sdiv_pipe"
-    ):
+        if prim == "seq_mem_d1":
+            if port_name == "read_en":
```

Comment:
```suggestion
        if prim == "seq_mem_d1" and port_name == "read_en":
```

Reply:
I was thinking this and then skipped over the simplification to emphasize the outer case, which adds a different set of ports (which happens to be a set of size 1)… hope that's cool!